Don't include malloc.h which is nonstandard and doesn't exist on macOS.
This source file doesn't use malloc but if it did, stdlib.h would be the
header to include for that.
--- tildepath.c.orig	2000-09-01 10:34:51.000000000 -0500
+++ tildepath.c	2022-01-08 14:16:44.000000000 -0600
@@ -38,7 +38,6 @@
 #include <stdio.h>
 #include <stdlib.h>
 #include <pwd.h>
-#include <malloc.h>
 #include <string.h>
 
 /*
